app/javascript/crayons/Modal/Modal.jsx

```diff
@@ -40,6 +41,10 @@ export const Modal = ({
   overlay,
   onClose,
 }) => {
+  const shortcuts = {
+    escape: onClose,
+  };
+
```

Comment:
If this PR is to close #12427, there's additional work we need to do - when a modal opens, the focus needs to be sent to the modal close button, and when the modal is closed, focus needs to be sent back to the link that activated it

Reply:
Agree - that's why I explicitly [didn't use `Closes` keyword](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) in the PR description since it's only part of the "requested" functionality.

I'm not sure how to _transfer_ focus so I didn't want to tackle it. If you have suggestions here I have my ears open :) Also feel free to commit too. Otherwise, I'll be ok with tackling focus transfer in a separate PR.

While typing this reply I took [focus-trap-react](https://github.com/focus-trap/focus-trap-react) package for a ride but for some reason it didn't work for me :c

Reply:
Ah cool, yep sorry I see it doesn't have the closes keyword 👍  Happy for this to be iterated on in a later PR for the focus management. There's currently a draft PR for adding focus-trap (#12366) so hopefully it'll all pull together soon!